contributor-docs/migrating-to-css-modules.md

```diff
@@ -0,0 +1,70 @@
+# Primer React Styled Components to CSS Modules Refactoring Guide
+
+This guide outlines the steps to follow when refactoring Primer React components from Styled Components to CSS Modules.
+
+---
+
+## Table of Contents
+
+- [Primer React Styled Components to CSS Modules Refactoring Guide](#primer-react-styled-components-to-css-modules-refactoring-guide)
+  - [Table of Contents](#table-of-contents)
+  - [Before Refactoring a Component](#before-refactoring-a-component)
+  - [How to Migrate](#how-to-migrate)
+    - [Refactoring Styled-Components to CSS Modules](#refactoring-styled-components-to-css-modules)
+    - [When Refactoring a Component](#when-refactoring-a-component)
+    - [Testing the Migration](#testing-the-migration)
+  - [Releasing a Component](#releasing-a-component)
+
+---
+
+## Before Refactoring a Component
+
+- **Verify VRT (Visual Regression Testing) Coverage:**
+  - Check for missing VRT coverage. We utilize the VRT tests to make sure we're matching styling in production with current expectations. Components should have a Storybook story for every "feature" or option available that impacts the UI for VRT to capture in a screenshot.
+  - Make sure there are `dev` stories for any edge cases spotted in production for the component (ie. sx prop, custom className, styled system attributes). `dev` stories may include things that we wouldn't normally recommend for the purpose of stress testing what happens when PRC components are overridden with custom styles. 
+- **Ensure All Visual Changes Are Completed:**
+  - Make necessary visual changes **before** creating the CSS Modules refactor PR.
+
+---
+
+## How to Migrate
+
+### Refactoring Styled-Components to CSS Modules
+
+- **Replace `${get('...')}` Syntax:**
+  - Migrate these to CSS Modules classes.
+  - No need for fallbacks in CSS Modules.
+
+### When Refactoring a Component
+
+1. **Check for `className` and `style` Prop:**
+   - Ensure the component accepts a `className` *on the top dom level only* for styling from outside of primer/react.
+   - Ensure the component accepts a `style` prop for more dynamic styling like positioning.
+2. **Feature Flagging:**
+   - Add a feature flag to toggle the `sx` prop for controlled rollout (staff shipping).
```

Comment:
I think it might be helpful to add a snippet showing what the feature flag code looks like for toggling the sx prop. Or even link to a previous PR that shows what a total migration might look like.